### PR TITLE
Add ebpf C linter

### PR DESCRIPTION
**Pull request by msherif1234**
remove deperceated pkg and update golanglint-ci
and ofcourse fixed new lint errors/warnings
recompiled ebpf files after new fmt changes

### CODE REVIEW

**Comment by msherif1234** [comment]
needs https://github.com/openshift/release/pull/50950

File: controllers/ingressnodefirewallconfig_controller_test.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package controllers
 import (
 	"context"
 	"fmt"
+	"k8s.io/utils/ptr"
```
Comment (frobware) [inline controllers/ingressnodefirewallconfig_controller_test.go:6]: Reorder to have standard library imports first, then k8s? It seems it was moved from a previous "k8s" grouping.

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,7 @@
 module github.com/openshift/ingress-node-firewall
 
-go 1.21
+go 1.21.0
```
Comment (frobware) [inline go.mod:3]: I've seen this happen on a few go mod changes recently. If you have `toolchain` do you actually need to be so specific with `go 1.21.0` or would `go 1.21` suffice?

File: pkg/interfaces/interfaces_test.go [hunk 3]
```diff
@@ -81,7 +81,9 @@ func TestGetInterfaceIndices(t *testing.T) {
 		t.Fatal(err)
 	}
 	t.Logf("bond interface index %d", bond.Attrs().Index)
-	defer netlink.LinkDel(bond)
+	defer func() {
+		_ = netlink.LinkDel(bond)
+	}()
```
Comment (frobware) [inline pkg/interfaces/interfaces_test.go:86]: If this is purely a stylistic change which continues to ignore the error from `LinkDel` is there actually any value?

File: test/e2e/namespaces/namespaces.go [hunk 4]
```diff
@@ -63,7 +63,7 @@ func CleanPods(namespace string, cs *testclient.ClientSet) error {
 		return nil
 	}
 	err := cs.Pods(namespace).DeleteCollection(context.Background(), metav1.DeleteOptions{
-		GracePeriodSeconds: pointer.Int64Ptr(0),
+		GracePeriodSeconds: ptr.To[int64](0),
```
Comment (frobware) [inline test/e2e/namespaces/namespaces.go:66]: Ooh! Generics! :)

**Review by frobware — CHANGES_REQUESTED** [review]
Generally LGTM. Why did the two eBPF object files change?

File: go.mod (hunk 2 above)
Comment (msherif1234) [reply go.mod:3]: `make vendor` is the one changing it for me

File: pkg/interfaces/interfaces_test.go (hunk 3 above)
Comment (msherif1234) [reply pkg/interfaces/interfaces_test.go:86]: wanted to keep linter quiet but I will add check and err log here while I am here

**Comment by msherif1234** [comment]
> Generally LGTM. Why did the two eBPF object files change?

I guess clang isn't smart enough to realize the diffs are all just white space

**Comment by frobware** [comment]
> Generally LGTM. Why did the two eBPF object files change?

... via `make ebpf-generate`.

File: pkg/ebpfsyncer/ebpfsyncer.go [hunk 5]
```diff
@@ -92,7 +92,7 @@ func (e *ebpfSingleton) SyncInterfaceIngressRules(
 		// Wait for a SIGTERM
 		<-c
 		if e.c != nil {
-			e.resetAll()
+			_ = e.resetAll()
```
Comment (frobware) [inline pkg/ebpfsyncer/ebpfsyncer.go:95]: Should any return value here be logged?

Comment (msherif1234) [reply pkg/ebpfsyncer/ebpfsyncer.go:95]: not really we are handling `SIGTERM` the container is going away at this point so its best effort cleanup `resetAll` will generate enough logs for error cases

**Pull request merged**